src/sql/SqlResult.ts

```diff
@@ -312,15 +319,10 @@ export class SqlResultImpl implements SqlResult {
      */
     fetch(): Promise<SqlPage> {
         // If there is an ongoing fetch, return that promise
-        if (this.fetchDeferred?.promise)
-            {return this.fetchDeferred.promise;}
-
-        // Do not start a fetch if close is called
-        if (this.closeDeferred?.promise) {
-            return Promise.reject(
-                new HazelcastSqlException(this.clientUUID, SqlErrorCode.CANCELLED_BY_USER, 'Query was cancelled by user')
-            );
+        if (this.fetchDeferred?.promise) {
+            return this.fetchDeferred.promise;
         }
+
```

Comment:
probably unnecessary will delete this

Reply:
do you mean the `
        if (this.fetchDeferred?.promise) {            return this.fetchDeferred.promise;        }` If it is, I think it can stay. This is needed for concurrent fetch requests, which should be possible with the API we have (the `next` method)

Reply:
no just the newline :)